app/javascript/lib/scp-api.js

```diff
@@ -260,10 +260,22 @@ export default async function scpApi(path, init, mock=false) {
   let fullPath = basePath + path
   if (mock) fullPath += '.json' // e.g. /mock_data/search/auth_code.json
 
+  function handleErrors(response) {
+    if (!response.ok) {
+      return Promise.reject(
+        response
+      )
+    }
+    return response
+  }
   const response = await fetch(fullPath, init)
-  const json = await response.json()
-
-  // Converts API's snake_case to JS-preferrable camelCase,
-  // for easy destructuring assignment.
-  return camelcaseKeys(json)
+    .then(handleErrors)
+    .catch(error => error)
+  if (response.ok) {
+    const json = await response.json()
```

Comment:
I would also put error handling around this response.json call.  A common failure case for rails APIs is an error in a part of a controller, that due to incomplete configuration, then returns an html error page rather than json.  In which case I believe this call will fail with an 'unexpected character, failed to parse' error